### PR TITLE
adding microsoft-calendar plugin

### PR DESCRIPTION
**Pull request by Abhay-soni-developer**
Signed-off-by: Abhay-soni-developer <abhaysoni.developer@gmail.com>

## Hey, I just made a Pull Request!

https://user-images.githubusercontent.com/23618736/214025699-95568746-8455-4749-8b1d-b2ff52fc2531.mp4

### CODE REVIEW

**Comment by Abhay-soni-developer** [comment]
@benjdlambert @tudi2d 
>> yarn run tsc is failing on files that do not contain any changes , what to do for this.

**Comment by awanlin** [comment]
Hi @Abhay-soni-developer, this is a pretty nice contribution, thanks! I'm not 100% sure what the issue is but I would try and rebase your branch, it's a bit behind and maybe that's what's causing the issues.

File: .github/CODEOWNERS
Comment (awanlin) [inline .github/CODEOWNERS:62]: Without actually getting write access to the repo we've found that adding to this file for external contributors won't do anything so I'm not sure it make sense to make this change.

File: plugins/microsoft-calendar/README.md
Comment (awanlin) [inline plugins/microsoft-calendar/README.md:1]: You'll want to update this with the proper content. Ideally a screenshot of the functionality so people can see what it does along with instructions as to how to install the plugin. Maybe look at this one for inspiration: https://github.com/backstage/backstage/tree/master/plugins/azure-devops

File: plugins/microsoft-calendar/src/components/CalendarCard.tsx
Comment (awanlin) [inline plugins/microsoft-calendar/src/components/CalendarCard.tsx:42]: Can you remove the commented out code in this file if it's not going to be used?

File: plugins/microsoft-calendar/src/components/CalendarEvent.tsx
Comment (awanlin) [inline plugins/microsoft-calendar/src/components/CalendarEvent.tsx:24]: Same here, if this isn't going to be used let's take it out 👍

File: plugins/microsoft-calendar/README.md
Comment (awanlin) [reply plugins/microsoft-calendar/README.md:1]: I'd also include lots of details around the auth part of this, is there anything that needs to be setup before hand to make this work. Just mentioning this as it looks like we rely in the Microsoft Auth, which makes sense, but we need to make sure users of the plugin are aware of this

**Review by awanlin — CHANGES_REQUESTED** [review]
This is pretty great! Took a dive through the code and left a few comments, nothing major. Will leave that to the Backstage Maintainers.

One thing to note is that we can not use any Microsoft Icons. You have the Outlook and Teams icons in this PR and we can't use them without permission from Microsoft. For reference: https://www.microsoft.com/en-us/legal/intellectualproperty/trademarks. You can use any of these as a replacement: https://v4.mui.com/components/material-icons/

**Comment by awanlin** [comment]
> Hi @Abhay-soni-developer, this is a pretty nice contribution, thanks! I'm not 100% sure what the issue is but I would try and rebase your branch, it's a bit behind and maybe that's what's causing the issues.

OK, I pulled down your branch and did a rebase. No luck, I'm getting the same errors after I run `yarn install` and then `yarn tsc`. I played around with the `yarn.lock` file as well but nothing resolve the errors. I have two PRs that I updated today and the builds all passed so I'm not sure what's going on here :(

File: .github/CODEOWNERS
Comment (benjdlambert) [reply .github/CODEOWNERS:62]: Even though it doesn't work as intended it's still useful to us so we can see the owner easily :) We have also spoken about writing some tooling ourselves to ping the relevant code owners from a bot account in a comment for notifications rather than permissions.

Comment (awanlin) [reply .github/CODEOWNERS:62]: Ah, that makes sense to me 🚀

**Comment by benjdlambert** [comment]
Think these errors are caused by the fact that the popup dependency depends on `muiv5` which is not currently supported, we would need a library that uses `muiv4` https://github.com/backstage/backstage/pull/15896/files#diff-51e4f558fae534656963876761c95b83b6ef5da5103c4adef6768219ed76c2deR28847

**Comment by Abhay-soni-developer** [comment]
@benjdlambert  @awanlin  @tudi2d i have made resolved all the pipeline error and made the necessary changes as per review give by you guys. 
i have opened a new pull request rather than continuing on this one as it gave me a clean slate to go step by step and resolve the errors that i was facing.

This is the new PR : https://github.com/backstage/backstage/pull/16060

Please do the needful

**Comment by benjdlambert** [comment]
@Abhay-soni-developer not sure the PR's are doing what you think they're doing 😅 The other PR doesn't seem to have only your commits in there? and is it a continuation of this or is it meant to replace this PR? Think it's going to be easier to have it as one PR rather than two separate ones. If you can close one of them, and then we can review it further.

It would be great if you can make this PR the final PR as this has the comment history from the review so we don't have to go back and forth :)

**Comment by Abhay-soni-developer** [comment]
> @Abhay-soni-developer not sure the PRs are doing what you think they're doing 😅 The other PR doesn't seem to have only your commits in there. and is it a continuation of this or is it meant to replace this PR? Think it's going to be easier to have it as one PR rather than two separate ones. If you can close one of them, and then we can review it further.
> 
> It would be great if you can make this PR the final PR as this has the comment history from the review so we don't have to go back and forth :)

I do understand your concern but I have also made the changes that you asked for in the feedback in that pr, so I would be closing this one you can review https://github.com/backstage/backstage/pull/16060 this pr by selecting commits range if you can, sorry for the hassle. I hope we can still get the mention PR merge as early as possible

**Pull request closed without merging**
